protocol/directory.go

```diff
@@ -70,13 +74,13 @@ func (d *ConiksDirectory) Register(req *RegistrationRequest) (
 		return nil, ErrorDirectory
 	}
 	if bytes.Equal(ap.LookupIndex, ap.Leaf.Index) {
-		return NewRegistrationProof(ap, d.LatestSTR(), nil), ErrorNameExisted
+		return NewRegistrationProof(ap, d.LatestSTR(), nil, ErrorNameExisted)
```

Comment:
As I said in https://github.com/coniks-sys/coniks-go/pull/71/commits/89cfe98c5ee06764e6855cdfbbc148a3d2f6f3ee#r75041201, I don't think we should be returning the error in this case since ``ErrorNameExisted`` will be propagated all the way up `listener.acceptClient()`, which I think should only be concerned about internal server errors (and maybe panic on those).